### ISSUE TITLE
[a11y]: JAWS does not announce the `kind` props of MenuItem

### ISSUE THREAD

**Issue by keijim2** [carbon-design-system/carbon#19286]
### Package

carbon-components

### Browser

Chrome

### Operating System

Windows

### Package version

v1.81.0

### React version

v18

### Automated testing tool and ruleset

n/a

### Assistive technology

JAWS

### Description

When focus is on `OverflowMenu` item with `kind="danger"`, background color is changed to "red". But it's color information is not announced by JAWS. The problem occurs with `MenuButton`, too.

![Image](https://github.com/user-attachments/assets/6728352d-de1b-4e91-95c0-c81d3aa99482)

The problem is reproducible on the followin storybook:
- https://react.carbondesignsystem.com/?path=/story/components-overflowmenu--default
- https://react.carbondesignsystem.com/?path=/story/components-menubutton--with-danger

In case of using `Button`, JAWS announce the kind of Button. As same as the `Button`, kind information for Menu Item should be announced by JAWS.
https://react.carbondesignsystem.com/?path=/story/components-button--danger

### WCAG 2.1 Violation

1.4.1 Use of Color

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-overflowmenu--default

### Steps to reproduce

In the following storybook,
1. Open context menu.
2. Press down-arrow key to move focus to "Delete app"
  -> JAWS does not read out information of "kind".

https://react.carbondesignsystem.com/?path=/story/components-overflowmenu--default

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by wkeese** [comment]
One could argue that the UI violates https://www.ibm.com/able/requirements/requirements/#1_4_1 by using color as the only way to convey information.   So you might think of this as a bug in [Carbon's guidance](https://carbondesignsystem.com/components/menu/usage/#danger-state) more so than a bug in @carbon/react itself. 
 (And if you did add an icon, then the icon could have an aria-label like "danger", and the problem is solved.)

If we don't want to add an icon, then we could solve the problem in this ticket (but not https://www.ibm.com/able/requirements/requirements/#1_4_1) by adding a visually hidden image with a "danger" `aria-label` (or something like that).

The counterargument to all of this is that the "Delete app" label is enough.  The red color is a nice hint, but even without knowing the color, screen reader users know that the operation is dangerous because it's called "Delete app".

**Comment by wkeese** [comment]
FYI, the danger **button** looks like

```
<button class="cds--btn cds--btn--danger" type="button" aria-describedby="danger-description-«r2n»">
  <span id="danger-description-«r2n»" class="cds--visually-hidden">danger</span>
  Remove asset
</button>
```

If we followed the same pattern, `kind=danger` on a `MenuItem` would do the same thing.

**Comment by wkeese** [comment]
@maradwan26 - You fixed `MenuItem` but not `OverflowMenuItem`.  Should we reopen this?

I know the next version of `OverflowMenu` (in OverflowMenu/next/index.tsx, available now via the `enable-v12-overflowmenu` feature flag) gets rid of `OverflowMenuItem` and just uses `MenuItem`, so I guess that's the other approach.

**Comment by maradwan26** [comment]
@wkeese Thanks for pointing this out. [Another PR is up](https://github.com/carbon-design-system/carbon/pull/21470) to address this for OverflowMenuItem
